appstore/developer_portal_api.py

```diff
@@ -0,0 +1,558 @@
+            "title": str,
+            "description": str,
+            "category": str,
+            "website": str,
+            "source": str,
+            "visible": bool
+        }
+
+        # Check all passed fields are allowed
+        for x in req:
+            if not type(x) == allowed_fields_type_map[x]:
+                return jsonify(error=f"Invalid value for field '{x}'", e=f"invalid.field.{x}"), 400
+
+
+        # Check app exists
+        try:
+            app = App.query.filter(App.id == app_id).one()
+        except NoResultFound:
+            return jsonify(error="Unknown app", e="app.notfound"), 400
+
+        # Check we own the app
+        if not is_users_developer_id(app.developer_id):
+            return jsonify(error="You do not have permission to modify that app", e="permission.denied"), 403
+
+        # Check any enum fields
+        if "category" in req and not is_valid_category(req["category"]):
+            return jsonify(error="Invalid value for field: category", e="invalid.field.category"), 400
+        if "visible" in req and not (req["visible"].lower() == "true" or req["visible"].lower() == "false"):
+            return jsonify(error="Invalid value for field: visible", e="invalid.field.visible"), 400
+
+        # Disallow change face category
+        if "category" in req and app.category == "Faces":
+            return jsonify(error="Cannot change category for watchface", e="disallowed.field.category"), 400
+
+        # Check title length
+        if "title" in req and len(req["title"]) > 45:
+            return jsonify(error="Title must be less than 45 characters", e="invalid.field.title"), 400
+            
+        # Update the app
+        for x in req:
+            setattr(app, x, req[x])
+
+        # Updating description requires iterating through asset collection
+        if "description" in req:
+            for x in app.asset_collections:
+                app.asset_collections[x].description = req["description"]
+
+        db.session.commit()
+        if algolia_index:
+            algolia_index.partial_update_objects([algolia_app(app)], { 'createIfNotExists': False })
+
+        return jsonify(success=True, id=app.id)
+
+
+@devportal_api.route('/app/<app_id>/release', methods=['POST'])
+def submit_new_release(app_id):
+    try:
+        app = App.query.filter(App.id == app_id).one()
+    except NoResultFound:
+        return jsonify(error="Unknown app", e="app.notfound"), 400
+
+    # Check we own the app
+    if not is_users_developer_id(app.developer_id):
+        return jsonify(error="You do not have permission to modify that app", e="permission.denied"), 403
+
+    data = dict(request.form)
+
+    if "pbw" not in request.files:
+        return jsonify(error="Missing file: pbw", e="pbw.missing"), 400
+
+    if "release_notes" not in data:
+        return jsonify(error="Missing field: release_notes", e="release_notes.missing"), 400
+
+    pbw_file = request.files['pbw'].read()
```

Comment:
same as Katharine's comment above, I think, re: zip files being passed directly into the thing?